framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2363,7 +2363,13 @@ protected void checkTypecastSafety(TypeCastTree typeCastTree) {
       return;
     }
     AnnotatedTypeMirror castType = atypeFactory.getAnnotatedType(typeCastTree);
+    if (!atypeFactory.canBeAnnotated(castType)) {
```

Comment:
It just seems like this (and the other calls to `canBeAnnotated`)  is a workaround for the defaulting rules in the Signedness Checker that aren't working correctly.